### ISSUE TITLE
Typescript doesn't like the types shipped with the module

### ISSUE THREAD

**Issue by RobertWHurst** [micromatch/anymatch#29]
The index.d.ts doesn't seem to play nice with Typescript v3.4.1. It results in the following errors:

![Screen Shot 2019-04-16 at 3 58 13 PM](https://user-images.githubusercontent.com/178857/56249468-bcab9880-6060-11e9-98bc-cd3837a0d269.png)

```
node_modules/anymatch/index.d.ts:5:20 - error TS7051: Parameter has a name but no type. Did you mean 'arg0: string'?

5 type AnymatchFn = (string) => boolean;
                     ~~~~~~

node_modules/anymatch/index.d.ts:8:73 - error TS2314: Generic type 'Array<T>' requires 1 type argument(s).

8 declare function anymatch(matchers: AnymatchMatcher, testString: string|Array, returnIndex?: boolean): boolean;
                                                                          ~~~~~

node_modules/anymatch/index.d.ts:9:75 - error TS2314: Generic type 'Array<T>' requires 1 type argument(s).

9 declare function anymatch(matchers: AnymatchMatcher): (testString: string|Array, returnIndex?: boolean) => number;
                                                                            ~~~~~

```

**Comment by paulmillr** [comment]
Could you try using github version: `npm i github:micromatch/anymatch` and test if that fixes your issue?

**Comment by chyzwar** [comment]
The simmilar issue with webpack typdef that depend on anymatch. 
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/webpack/index.d.ts

**Comment by paulmillr** [comment]
We've fixed this shit, you need to update

**Comment by chyzwar** [comment]
Currently released version 3.0.1 do not expose Matcher type https://github.com/micromatch/anymatch/blob/3.0.1/index.d.ts

Master contain fix but It was not yet released.
https://github.com/micromatch/anymatch/blob/master/index.d.ts

**Comment by apiel** [comment]
When do you plan to release the bugfix?
